pylanetary/spectral/core.py

```diff
@@ -0,0 +1,808 @@
+  #test_wind, test_err = calc_doppler_vel(peak = test_peak, err = test_std, rest_f = 345.79598990)
+  
+###############################################################################
+###############################################################################
+  
+class SpectralCube:
+  '''
+  Analysis tools for a FITS image cube.
+  
+  Example Usage, full image analysis:
+  
+  cube = SpectralCube(your_image_path, x_space = 'f')
+  
+  data = cube.extract_image()
+  cube.fit_data(data, fit_type = 'Moffat', outfile = 'development_testing_fit', RMS = 0.017530593, SN = 6,initial_fit_guess = '/homes/metogra/skathryn/Research/Scripts/co_moffat_modelresult.sav')
+  test_cube.wind_calc(picklefile = 'development_testing_fit.pickle', restfreq = 345.79598990, outfile = 'development_testing_wind')
+  test_cube.plot_data(picklefile = 'development_testing_wind.pickle', platescale = 0.1, body = 'Neptune', title = 'CO Doppler Velocity Map', cont_label = 'Radial Velocity (m/s)', date = '2016-04-30 00:00', location = 'ALMA', spatial = 4e4, limits = [-2000,2000], contours = [21,200])
+  
+  '''
+  ###############################################################################
+  
+  def __init__(self, imagepath, x_space):
+    '''
+    Input string instantiates object using the FITS image path.
+    
+    Parameters
+    ----------
+    imagepath : string
+          The path to FITS image file.
+    x_space : string
+          The identifier for choosing wavelength or frequency space for the model fit. Input 'F' or 'f' for frequency space, 'W' or 'w' for wavelength space.
+    
+    Attributes
+    ----------
+    chan_num : float
+          The number of frequency channels. Pulled from image header.
+    i0 : float
+          The reference pixel of the image. Pulled from image header.
+    df : float
+          The channel width in Hz. Pulled from image header.
+    f0 : float
+          The rest frequency of the image in Hz. Pulled from image header.
+    xpixmax : float
+          The x-dimension size in pixels. Pulled from image header.
+    ypixmax : float
+          The y-dimension size in pixels. Pulled from image header.
+    bmaj : float
+          The semi-major axis of the restoring beam in arcsec. Pulled from image header. 
+    bmin : float
+          The semi-minor axis of the restoring beam in arcsec. Pulled from the image header.
+    bpa : float
+          The position angle of the restoring beam in degrees. Pulled from the image header.
+    data : numpy array
+          The spectral data pulled from the image cube.
+    freqspec : numpy array
+          The frequency grid around the rest frequency, which can be used as the x-axis.
+    wl_spec : numpy array
+          The wavelength grid around the rest wavelength, which can be used as the x-axis.
+    
+    '''  
+    self.x_space = x_space
+    
+    fits_file = fits.open(imagepath)
```

Comment:
In the future, we'd like to make it easy to read data from multiple different telescopes, and unfortunately, all of those telescopes use different fits conventions.  This is a constant annoyance of mine when trying to code stuff.  There are multiple ways to handle this and no easy solution.  For now, though, it would be helpful if you could move the lines below into a separate function (outside of the class) named something like load_fits_alma(filename)  And then call it like
`self.chan_num, self.i0, self.f0, ... self.data = load_fits_alma(filename)`
It's admittedly a bit ugly to have so many returns from a function, but it at least makes it obvious to a new user how to hack this code to support a new telescope - they would simply write load_fits_ngvla() or whatever with the same returns.